### PR TITLE
[ESM] Add table for feature-specific coverage

### PR DESCRIPTION
**Pull request by gregfurman**
## Changes

Adds a new coverage table for the Lambda event-source mapping section of the docs.

The new table will (approximately) look like:

  |   | SQS |   | Stream |   | Kafka ⭐️ |  
-- | -- | -- | -- | -- | -- | -- | --
Parameter | Description | Standard | FIFO | Kinesis | DynamoDB | Amazon MSK | Self-Managed
BatchSize | Batching events by count. | 🟡 | 🟢 | 🟢 | 🟢 | 🟢 | 🟢
➖ | Batch when ≥ 6 MB limit. | 🟠 | 🟠 | 🟠 | 🟠 | 🟢 | 🟢
MaximumBatchingWindowInSeconds | Batch by Time Window. | 🟠 | ➖ | 🟠 | 🟠 | 🟢 | 🟢
MaximumRetryAttempts | Discard after N retries. | ➖ | ➖ | 🟢 | 🟢 | ➖ | ➖
MaximumRecordAgeInSeconds | Discard records older than time t. | ➖ | ➖ | 🟢 | 🟢 | ➖ | ➖
Enabled | Enabling/Disabling. | 🟢 | 🟢 | 🟢 | 🟢 | 🟢 | 🟢
FilterCriteria | Filter criteria object. | 🟢 | 🟢 | 🟢 | 🟢 | 🟢 | 🟢
FunctionResponseTypes | Enabling ReportBatchItemFailures. | 🟢 | 🟢 | 🟢 | 🟢 | ➖ | ➖
BisectBatchOnFunctionError | Bisect a batch on error and retry. | ➖ | ➖ | 🟠 | 🟠 | ➖ | ➖
ScalingConfig | The scaling configuration for the event source. | 🟠 | 🟠 | ➖ | ➖ | ➖ | ➖
ParallelizationFactor | Parallel batch processing by shard. | ➖ | ➖ | 🟠 | 🟠 | ➖ | ➖
DestinationConfig.OnFailure | SQS Failure Destination. | ➖ | ➖ | 🟢 | 🟢 | 🟢 | 🟢
  | SNS Failure Destination. | ➖ | ➖ | 🟢 | 🟢 | 🟢 | 🟢
  | S3 Failure Destination. | ➖ | ➖ | 🟠 | 🟠 | 🟠 | 🟠
DestinationConfig.OnSuccess | Success Destinations. | ➖ | ➖ | ➖ | ➖ | ➖ | ➖
MetricsConfig | CloudWatch metrics. | 🟠 | 🟠 | 🟠 | 🟠 | 🟠 | 🟠
ProvisionedPollerConfig | Control throughput via min-max limits. | ➖ | ➖ | ➖ | ➖ | 🟠 | 🟠
StartingPosition | Position to start reading from. | ➖ | ➖ | 🟢 | 🟢 | 🟢 | 🟢
StartingPositionTimestamp | Timestamp to start reading from. | ➖ | ➖ | 🟢 | ➖ | 🟢 | 🟢
TumblingWindowInSeconds | Duration (seconds) of a processing window. | ➖ | ➖ | 🟠 | 🟠 | ➖ | ➖
Topics ⭐️ | Kafka topics to read from. | ➖ | ➖ | ➖ | ➖ | 🟢 | 🟢

#### Legend
- ⭐️ Only Available in Pro
- 🟢 Fully Implemented
- 🟡 Partially Implemented
- 🟠 Not Implemented
- ➖ Not Applicable (Not Supported by AWS)

### CODE REVIEW

File: content/en/user-guide/aws/lambda/index.md [hunk 1]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
...
-- Only basic validations are performed upon creating and updating ESM.
-- Streaming Pollers for Kinesis and DynamoDB do not implement the following batching behavior configurations:
-  - `BisectBatchOnFunctionError`
-  - `MaximumBatchingWindowInSeconds`
-  - `ParallelizationFactor`
-  - `ScalingConfig`
-  - `TumblingWindowInSeconds`.
+- [Self-Managed Apache Kafka](https://docs.aws.amazon.com/lambda/latest/dg/with-kafka.html)
+
+### Coverage
```
Comment (gregfurman) [inline content/en/user-guide/aws/lambda/index.md:174]: From @joe4dev:

@HarshCasper How do we best present such a feature model to the user?

We are trying to improve the transparency regarding the supported behavioral features of Lambda Event Source Mapping beyond the simple list of supported sources we currently have in the Lambda docs [here](https://docs.localstack.cloud/user-guide/aws/lambda/#lambda-event-source-mappings). Should we just add the table below to the Lambda docs (similar than the [API coverage docs](https://docs.localstack.cloud/references/coverage/coverage_lambda/), but for behavior)

File: content/en/user-guide/aws/lambda/index.md [hunk 2]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
...
+- 🟢 Fully Implemented
+- 🟡 Partially Implemented
+- 🟠 Not Implemented
+- ➖ Not Applicable (Not Supported by AWS)
+
+| | <th colspan="2" style="text-align:center;">SQS</th> <th colspan="2" style="text-align:center;">Stream</th> <th colspan="2" style="text-align:center;">Kafka ⭐️</th>
+|--------------------------------|-------------------------------------------------|:--------:|:----:|:---------:|:----------:|:----------:|:------------:|
+| **Parameter**                  | **Description**                                 | **Standard** | **FIFO** | **Kinesis** | **DynamoDB** | **Amazon MSK** | **Self-Managed** |
+| BatchSize                      | Batching events by count.                       | 🟡 [^1]   | 🟢    | 🟢       | 🟢        | 🟢          | 🟢            |
+| ➖                             | Batch when ≥ 6 MB limit.                        | 🟠        | 🟠    | 🟠       | 🟠        | 🟢          | 🟢            |
```
[the remark this answers is not in the document]
Comment (gregfurman) [reply content/en/user-guide/aws/lambda/index.md]: Replaced with `*Not Configurable*`

File: content/en/user-guide/aws/lambda/index.md [hunk 3]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
...
-  - `BisectBatchOnFunctionError`
-  - `MaximumBatchingWindowInSeconds`
-  - `ParallelizationFactor`
-  - `ScalingConfig`
-  - `TumblingWindowInSeconds`.
+- [Self-Managed Apache Kafka](https://docs.aws.amazon.com/lambda/latest/dg/with-kafka.html)
+
+### Coverage
+
+The coverage table below shows feature-level coverage for all supported event sources for the latest version of LocalStack.
```
[the remark this answers is not in the document]
Comment (gregfurman) [reply content/en/user-guide/aws/lambda/index.md]: Added a sentance for this!

File: content/en/user-guide/aws/lambda/index.md [hunk 4]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
 [Lambda event source mappings](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventsourcemapping.html) allows you to connect Lambda functions to other AWS services.
 The following event sources are supported in LocalStack:
 
+- [Simple Queue Service (SQS)](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html)
 - [DynamoDB](https://docs.aws.amazon.com/lambda/latest/dg/with-ddb.html)
 - [Kinesis](https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html)
 - [Managed Streaming for Apache Kafka (MSK)](https://docs.aws.amazon.com/lambda/latest/dg/with-msk.html)
```
[the remark this answers is not in the document]
Comment (gregfurman) [reply content/en/user-guide/aws/lambda/index.md]: Done

File: content/en/user-guide/aws/lambda/index.md [hunk 5]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
...
+| FunctionResponseTypes          | Enabling ReportBatchItemFailures.               | 🟢        | 🟢    | 🟢       | 🟢        | ➖          | ➖            |
+| BisectBatchOnFunctionError     | Bisect a batch on error and retry.              | ➖        | ➖    | 🟠       | 🟠        | ➖          | ➖            |
+| ScalingConfig                  | The scaling configuration for the event source. | 🟠        | 🟠    | ➖       | ➖        | ➖          | ➖            |
+| ParallelizationFactor          | Parallel batch processing by shard.             | ➖        | ➖    | 🟠       | 🟠        | ➖          | ➖            |
+| DestinationConfig.OnFailure    | SQS Failure Destination.                        | ➖        | ➖    | 🟢       | 🟢        | 🟢          | 🟢            |
+|                                | SNS Failure Destination.                        | ➖        | ➖    | 🟢       | 🟢        | 🟢          | 🟢            |
+|                                | S3 Failure Destination.                         | ➖        | ➖    | 🟠       | 🟠        | 🟠          | 🟠            |
```
[the remark this answers is not in the document]
Comment (gregfurman) [reply content/en/user-guide/aws/lambda/index.md:207]: 🫡

**Review by joe4dev — APPROVED** [review]
Amazing 🤩 

Let's merge it and get some feedback 👂

File: content/en/user-guide/aws/lambda/index.md [hunk 6]
```diff
@@ -165,20 +165,51 @@ The product of 10 and 10 is 100%
...
+| BatchSize                      | Batching events by count.                       | 🟡 [^1]   | 🟢    | 🟢       | 🟢        | 🟢          | 🟢            |
+| ➖                             | Batch when ≥ 6 MB limit.                        | 🟠        | 🟠    | 🟠       | 🟠        | 🟢          | 🟢            |
+| MaximumBatchingWindowInSeconds | Batch by Time Window.                           | 🟠        | ➖    | 🟠       | 🟠        | 🟢          | 🟢            |
+| MaximumRetryAttempts           | Discard after N retries.                        | ➖        | ➖    | 🟢       | 🟢        | ➖          | ➖            |
+| MaximumRecordAgeInSeconds      | Discard records older than time `t`.            | ➖        | ➖    | 🟢       | 🟢        | ➖          | ➖            |
+| Enabled                        | Enabling/Disabling.                             | 🟢        | 🟢    | 🟢       | 🟢        | 🟢          | 🟢            |
+| FilterCriteria                 | Filter criteria object.                         | 🟢        | 🟢    | 🟢       | 🟢        | 🟢          | 🟢            |
```
[the remark this answers is not in the document]
Comment (gregfurman) [reply content/en/user-guide/aws/lambda/index.md]: For posterity: we added a footnote showing certain Metadata fields are unsupported as well as a link to the filter criteria docs on AWS.

**Pull request merged**